### PR TITLE
Adds model building tutorial

### PR DESCRIPTION
**Pull request by jmcook1186**
adds a tutorial for how to build a model plugin and load it into IF.

### CODE REVIEW

File: website/.docusaurus/docusaurus-plugin-content-docs/default/site-docs-06-tutorials-how-to-build-models-md-e64.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  "sidebar": "tutorialSidebar",
+  "previous": {
+    "title": "Tutorials",
+    "permalink": "/docs/tutorials/"
+  },
+  "next": {
+    "title": "IMPL writing guide",
+    "permalink": "/docs/tutorials/how-to-write-impls"
+  }
+}
```
Comment (narekhovhannisyan) [inline website/.docusaurus/docusaurus-plugin-content-docs/default/site-docs-06-tutorials-how-to-build-models-md-e64.json:24]: newline

File: website/.docusaurus/docusaurus-plugin-content-docs/default/site-docs-06-tutorials-how-to-write-impls-md-865.json [hunk 2]
```diff
@@ -17,7 +17,7 @@
   },
   "sidebar": "tutorialSidebar",
   "previous": {
-    "title": "Tutorials",
-    "permalink": "/docs/tutorials/"
+    "title": "Building model plugins",
+    "permalink": "/docs/tutorials/how-to-build-models"
   }
 }
```
Comment (narekhovhannisyan) [inline website/.docusaurus/docusaurus-plugin-content-docs/default/site-docs-06-tutorials-how-to-write-impls-md-865.json:23]: newline

File: website/.docusaurus/docusaurus-plugin-content-docs/default/site-docs-06-tutorials-index-md-12f.json [hunk 3]
```diff
@@ -18,7 +18,7 @@
     "permalink": "/docs/models/watt-time"
   },
   "next": {
-    "title": "IMPL writing guide",
-    "permalink": "/docs/tutorials/how-to-write-impls"
+    "title": "Building model plugins",
+    "permalink": "/docs/tutorials/how-to-build-models"
   }
 }
```
Comment (narekhovhannisyan) [inline website/.docusaurus/docusaurus-plugin-content-docs/default/site-docs-06-tutorials-index-md-12f.json:24]: newline

File: website/docs/06-tutorials/how-to-build-models.md [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+- create a Typescript file that implements the `IOutputModelInterface`
+- provide the path to the model to the IF command line tool.
+
+## The model interface
+
+The `IOutputModelInterface` is structured as follows:
+
+```ts
+export interface IOutputModelInterface {
+  modelIdentifier(): string;
```
Comment (narekhovhannisyan) [inline website/docs/06-tutorials/how-to-build-models.md:15]: you can exclude model identifier

File: website/docs/06-tutorials/how-to-build-models.md [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+
+  configure(
+    name: string,
```
Comment (narekhovhannisyan) [inline website/docs/06-tutorials/how-to-build-models.md:18]: it seems Gnanakeethan agreed that we can remove it, let's discuss

File: website/.docusaurus/docusaurus-plugin-content-docs/default/site-docs-06-tutorials-how-to-build-models-md-e64.json (hunk 1 above)
Comment (jmcook1186) [reply website/.docusaurus/docusaurus-plugin-content-docs/default/site-docs-06-tutorials-how-to-build-models-md-e64.json:24]:
```suggestion
}

```

File: website/.docusaurus/docusaurus-plugin-content-docs/default/site-docs-06-tutorials-how-to-write-impls-md-865.json (hunk 2 above)
Comment (jmcook1186) [reply website/.docusaurus/docusaurus-plugin-content-docs/default/site-docs-06-tutorials-how-to-write-impls-md-865.json:23]:
```suggestion
}

```

File: website/.docusaurus/docusaurus-plugin-content-docs/default/site-docs-06-tutorials-index-md-12f.json (hunk 3 above)
Comment (jmcook1186) [reply website/.docusaurus/docusaurus-plugin-content-docs/default/site-docs-06-tutorials-index-md-12f.json:24]:
```suggestion
}

```

**Pull request merged**